src/quickgui/qgsquickutils.cpp

```diff
@@ -27,10 +27,58 @@
 #include "qgsunittypes.h"
 
 
+QgsQuickUtils *QgsQuickUtils::sInstance = 0;
+
+QgsQuickUtils *QgsQuickUtils::instance()
+{
+  if ( !sInstance )
+  {
+    QgsDebugMsg( QStringLiteral( "QgsQuickUtils created: %1" ).arg( long( QThread::currentThreadId() ) ) );
+    sInstance = new QgsQuickUtils();
+  }
+  return sInstance;
+}
+
 QgsQuickUtils::QgsQuickUtils( QObject *parent )
   : QObject( parent )
-  , mScreenDensity( calculateScreenDensity() )
 {
+
+  // calculate screen density for calculation of real pixel sizes from density-independent pixels
+  int dpiX = QApplication::desktop()->physicalDpiX();
+  int dpiY = QApplication::desktop()->physicalDpiY();
+  int dpi = dpiX < dpiY ? dpiX : dpiY; // In case of asymmetrical DPI. Improbable
+  mScreenDensity = dpi / 160.;  // 160 DPI is baseline for density-independent pixels in Android
+}
+
+QgsQuickUtils::~QgsQuickUtils()
+{
+}
+
+QgsCoordinateReferenceSystem QgsQuickUtils::coordinateReferenceSystemFromEpsgId( long epsg ) const
```

Comment:
we do not need this, do we? For tests just create new QgsCoordinateReferenceSystem directly